compiler/rustc_lint/src/early.rs

```diff
@@ -446,12 +446,13 @@ pub fn check_ast_node<'a>(
             );
         }
     } else {
+        let pass_count = passes.len();
         for (i, pass) in passes.iter_mut().enumerate() {
             buffered =
                 sess.prof.extra_verbose_generic_activity("run_lint", pass.name()).run(|| {
                     early_lint_node(
                         sess,
-                        pre_expansion && i == 0,
+                        !pre_expansion && i == pass_count,
```

Comment:
when is `i == pass_count` true? `i` should be in the range `0..pass_count` :thinking:

also, can you add a comment on why that `i == X` check is needed?

Reply:
Good catch! The test works great locally but I suppose the other condition of the if is visited.

Reply:
Indeed `i == 0` works too, I've tried it. I've also added a revision to the test file so that it's actually being tested.